### ISSUE TITLE
Use valid-jsdoc to prohibit special annotations

### ISSUE THREAD

**Issue by ketamin** [eslint/eslint#7453]
Please be aware, I updated the whole description to match @nzakas improvement and also came up with an new idea.

Please consider this as an Change Request.

> What rule do you want to change?

valid-jsdoc

> Does this change cause the rule to produce more or fewer warnings?

Due to this is an feature-enrichment, I would consider, to have only more warnings, if user specifies this rule.

> How will the change be implemented? (New option, new default behavior, etc.)?

Extend the rule _valid-jsdoc_ with these two options:

``` json
"valid-jsdoc": ["warn", {
    "prohibit": [
        "public"
    ],
    "unknownAnnotations": false
}]
```

or maybe something, like this fits better:

``` json
"valid-jsdoc": ["warn", {
    "prohibit": {
        "unknownAnnotations": true,
        "public": true
    }
}]
```

Both should result in the exactly same way, I just wanted to provide two ideas, which one could better fit the concept is up to the maintainers, or the discussion below.

> Please provide some example code that this change will affect.

``` javascript
/**
 * @public
 * @myOwnStrangeAnnotation
 * @description Some usefull text.
 */
```

> What does the rule currently do for this code?

valid-jsdoc currently doesn't have an option, to show users useless (@public) or unknown annotations (@myOwnStrangeAnnotation).
The user may end up in having typos within the annotations, which, at least in my environment, will be shown, while jsdoc is generated. I ended up using ["allowUnknownTags":false](http://usejsdoc.org/about-configuring-jsdoc.html#tags-and-tag-dictionaries) to force usefull annotations, but often users complain to me, why this isn't done while typing.

> What will the rule do after it's changed?

The first two lines of the sample code would each result in a warning.

Hopefully I am not the only one living in a world of strange annotations popping out of nowhere ;-)

**Comment by not-an-aardvark** [comment]
Thanks for the proposal. This seems like a reasonable addition to me. :+1:

**Comment by nzakas** [comment]
@ketamin can you please update your description to use this template: https://github.com/eslint/eslint/blob/master/templates/rule-change-proposal.md

**Comment by ketamin** [comment]
Thanks @nzakas, I updated the description and re-thought everything, hopefully matching more peoples needs.

**Comment by kaicataldo** [comment]
@ketamin Sorry, seems we lost track of this - thanks for updating the description! @eslint/eslint-team Thoughts?

**Comment by mikesherov** [comment]
:+1:

**Comment by vitorbal** [comment]
Seems reasonable to me too :+1:

**Comment by nzakas** [comment]
@mikesherov just a reminder to leave your reaction on the issue description for it to count. 

@ketamin how do you propose that we track which tags are unknown?

I'm also not sure this belongs as part of valid-jsdoc, so I'd like to here some thoughts on why this shouldn't be a separate rule. On the surface, valid-JSDoc is about matching function signatures to the JSDoc comments, so I'm having a hard time understanding why warning on certain tags belongs here.

**Comment by mikesherov** [comment]
@nzakas all this time I've been replying to issues from the mobile interface, which is reactionless! 🤕

**Comment by ilyavolodin** [comment]
I think I like the idea of `prohibit`, but dislike the idea of `unknownAnnotations`. As @nzakas noted, we don't have a way to know which tags are unknown, and the only way we can track that is by creating and maintaining a list (which we would need to update, every time doctrine adds support for a new tag). I think `unknownAnnotations` belongs to [eslint-plugin-jsdoc](https://github.com/gajus/eslint-plugin-jsdoc), but I can support adding `prohibit` with exact string matching.

**Comment by ketamin** [comment]
@nzakas and @ilyavolodin Thanks, didn't know the eslint-plugin-jsdoc. But it has the `unknownAnnotations` as [check-tag-names](https://github.com/gajus/eslint-plugin-jsdoc#eslint-plugin-jsdoc-rules-check-tag-names). So I totally agree in not going any further to this one.

**Comment by vitorbal** [comment]
Unless someone is willing to implement the `prohibit` option, I think we can close this.

**Comment by alberto** [comment]
Thanks for your interest in improving eslint. Unfortunately, it looks like this issue didn't get enough support from the team and so I'm closing it. While we wish we'd be able to accommodate everyone's requests, we do need to prioritize. We've found that issues failing to reach consensus after 21 days tend to never do it, and as such, we close those issues. This doesn't mean the idea isn't interesting, just that it's not something the team can commit to.
